libeos-payg/multi-task.h

```diff
@@ -30,6 +30,9 @@ void epg_multi_task_attach         (GTask         *task,
 void epg_multi_task_increment      (GTask         *task);
 void epg_multi_task_return_boolean (GTask         *task,
                                     gboolean       result);
+void epg_multi_task_return_pointer (GTask         *task,
```

Comment:
Note: As discussed in person in London, it would be good if `EpgMultiTask` could be submitted upstream to GLib. However, that’s definitely not a prerequisite for this PR. I’m not going to do API review on it here, since the API seems appropriate for its use in this PR.

Upstream-wise, I think the API we discussed was to have the `MultiTask` expose all the return values from all its sub-tasks, and allow the caller to implement their own result accumulation; rather than trying to provide some generic mechanism for result accumulation.